frontend/src/routes/(authenticated)/project/create/+page.svelte

```diff
@@ -85,6 +89,45 @@
   $: $asyncCodeError = $codeIsAvailable ? undefined : $t('project.create.code_exists');
   const codeErrors = derived([errors, asyncCodeError], () => [...new Set(concatAll($errors.code, $asyncCodeError))]);
 
+  const projectNameStore = derived(form, ($form) => $form.name);
+  const langCodeStore = derived(form, ($form) => $form.languageCode);
+  const orgIdStore = derived(form, ($form) => $form.orgId);
+  const langCodeAndOrgIdStore: Readable<{langCode: string, orgId: string}> = derived([langCodeStore, orgIdStore], ([lang, orgId], set) => {
+    if (lang && orgId && (lang.length == 2 || lang.length == 3)) {
+      set({ langCode: lang, orgId: orgId });
+    }
+  });
+
+  const projectNameAndOrgIdStore: Readable<{projectName: string, orgId: string}> = derived([projectNameStore, orgIdStore], ([projectName, orgId], set) => {
+    if (projectName && orgId && projectName.length >= 3) {
+      set({ projectName, orgId: orgId });
+    }
+  });
+
+  const relatedProjectsByLangCodeStoreStore = deriveAsyncIfDefined(langCodeAndOrgIdStore, _getProjectsByLangCodeAndOrg);
+  const relatedProjectsByNameStoreStore = deriveAsyncIfDefined(projectNameAndOrgIdStore, _getProjectsByNameAndOrg);
```

Comment:
This is the only part of the code that I still don't like much and hope to improve. I originally tried to write `deriveAsyncIfDefined` in such a way that it wouldn't return a "store store" (a `Readable<Readable<T>>`) and couldn't figure out how to do it, which is why this bit, plus the stores on lines 117 and 121, are kind of ugly. I might at least be able to write an `unwrapNestedStore<T>` function that will turn `Readable<Readable<T>>` into `Readable<T>`, and that might help clean up some of this ugliness.

Reply:
Commit 0ba3b73 makes this much nicer.